dot/state/block.go

```diff
@@ -250,6 +250,37 @@ func (bs *BlockState) GetHashByNumber(num uint) (common.Hash, error) {
 	return common.NewHash(bh), nil
 }
 
+// GetBlocksBySlot gets all blocks that were produced in given slot.
+func (bs *BlockState) GetBlocksBySlot(slotNum uint64) ([]common.Hash, error) {
+	highestFinalisedHash, err := bs.GetHighestFinalisedHash()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get highest finalised hash: %w", err)
+	}
+
+	descendants, err := bs.bt.GetAllDescendants(highestFinalisedHash)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get descendants of %s: %w", highestFinalisedHash, err)
+	}
+
+	blocksWithGivenSlot := []common.Hash{}
+
+	for _, desc := range descendants {
+		descSlot, err := bs.GetSlotForBlock(desc)
+		if errors.Is(err, types.ErrGenesisHeader) {
+			continue
+		}
+		if err != nil {
+			return blocksWithGivenSlot, fmt.Errorf("could not get slot for block %s: %w", desc, err)
```

Comment:
> to avoid repeating it

desc gets printed only once in this case. I would rather add that context here once than adding it in all errors in `GetSlotForBlock`